sdk/spring/azure-spring-integration-servicebus/src/test/java/com/azure/spring/integration/servicebus/ServiceBusMessageConverterTest.java

```diff
@@ -103,4 +104,51 @@ public void shouldConvertSpringMessageHeaderIntoIMessage() {
         assertNotNull(servicebusMessage.getScheduledEnqueueTimeUtc());
 
     }
+
+    @Test
+    public void messageIdTest() {
+        org.springframework.messaging.Message<String> springMessage;
+        IMessage serviceBusMessage;
+        org.springframework.messaging.Message<String> convertedSpringMessage;
+
+        // springMessage can not set header: MessageHeaders.ID
+        // case 1: RAW_ID not set.
+        springMessage = MessageBuilder.withPayload(payload).build();
+        assertTrue(springMessage.getHeaders().containsKey(MessageHeaders.ID)); // spring message always have an id.
+        assertFalse(springMessage.getHeaders().containsKey(AzureHeaders.RAW_ID));
+
+        serviceBusMessage = getConverter().fromMessage(springMessage, IMessage.class);
+        assertTrue(serviceBusMessage.getProperties().containsKey(MessageHeaders.ID));
+        assertFalse(serviceBusMessage.getProperties().containsKey(AzureHeaders.RAW_ID));
+        assertNotNull(serviceBusMessage.getMessageId());
+        assertEquals(springMessage.getHeaders().get(MessageHeaders.ID).toString(), serviceBusMessage.getMessageId());
+
+        convertedSpringMessage = getConverter().toMessage(serviceBusMessage, String.class);
+        assertTrue(convertedSpringMessage.getHeaders().containsKey(MessageHeaders.ID));
+        assertTrue(convertedSpringMessage.getHeaders().containsKey(AzureHeaders.RAW_ID));
+        assertNotEquals(springMessage.getHeaders().get(MessageHeaders.ID), convertedSpringMessage.getHeaders().get(MessageHeaders.ID)); // spring message id always not equal to another message. See MessageHeaderAccessor#isReadOnly() for more detail
+        assertNotEquals(serviceBusMessage.getMessageId(), convertedSpringMessage.getHeaders().get(MessageHeaders.ID));
+        assertEquals(serviceBusMessage.getMessageId(), convertedSpringMessage.getHeaders().get(AzureHeaders.RAW_ID)); // convertedSpringMessage always have header: AzureHeaders.RAW_ID, equal to serviceBusMessage.getMessageId().
```

Comment:
Maybe customer will be confused about this: convertedSpringMessage always have header: AzureHeaders.RAW_ID. But we can not remove it. Because we don't know whether it's come set from sender(by RAW_ID) or spring-messaging generated.